core/src/apps/cardano/auxiliary_data.py

```diff
@@ -0,0 +1,255 @@
+from trezor.crypto import hashlib
+from trezor.crypto.curve import ed25519
+from trezor.messages import (
+    CardanoAddressType,
+    CardanoAuxiliaryDataType,
+    CardanoMetadataType,
+)
+
+from apps.common import cbor
+
+from ..common.seed import remove_ed25519_prefix
+from .address import derive_address_bytes, derive_human_readable_address
+from .helpers import INVALID_AUXILIARY_DATA, bech32
+from .helpers.bech32 import HRP_PUBLIC_KEY
+from .helpers.paths import SCHEMA_STAKING_ANY_ACCOUNT
+from .layout import confirm_catalyst_registration, show_auxiliary_data_hash
+
+if False:
+    from typing import Dict, List, Optional, Tuple, Union
+    from trezor import wire
+
+    from trezor.messages.CardanoCatalystRegistrationParametersType import (
+        CardanoCatalystRegistrationParametersType,
+    )
+    from trezor.messages.CardanoTxAuxiliaryDataType import CardanoTxAuxiliaryDataType
+
+    CatalystRegistrationBody = Dict[int, Union[bytes, int]]
+    CatalystRegistrationBodyMap = Dict[int, CatalystRegistrationBody]
+    CatalystRegistrationSignatureMap = Dict[int, bytes]
+    SignedVotingRegistrationMap = Dict[
+        int, Union[CatalystRegistrationBody, CatalystRegistrationSignatureMap]
+    ]
+
+    from . import seed
+
+AUXILIARY_DATA_HASH_SIZE = 32
+PUBLIC_KEY_LENGTH = 32
+MAX_AUXILIARY_DATA_LENGTH = 500
```

Comment:
wasn't this limit the one we internally discussed to be no longer relevant?

Reply:
Yep, I removed the check, forgot to remove the constant. Constant removed - https://github.com/vacuumlabs/trezor-firmware/pull/57/commits/44d2859daa4e72b2f5a4ac327c021b62a89713ea